### PR TITLE
Refactor: remove role assignment from Terraform

### PR DESCRIPTION
**Pull request by angela-tran**
Part of #296 

We found that because the pipeline service principal defaults to having `Contributor` role, it is not able to create the role assignment for the ETL principal on the Storage Container. 

After some discussion, we decided we should remove the role assignment from being managed by Terraform / the pipeline, and it'll be a part of the manual creation steps to set up the role assignment.

This PR deletes the role configuration from Terraform and adds documentation for the manual steps.

### Pre-merge / post-merge actions
For **MST**, we need to 
 - [x] run `terraform state rm azurerm_role_assignment.velocity_etl[0]` so that Terraform "forgets" about the role and doesn't try to destroy it (see [docs on `rm`](https://developer.hashicorp.com/terraform/cli/commands/state/rm)). 
 - After merging, the `prod` pipeline should succeed, and there's nothing left to do.

For **SBMTD**, 
 - [x] there's no pre-merge action to do. 
 - After merging, the `prod` pipeline should succeed since now there is no permissions error to run into. The last thing to do is to manually create the role assignment for the ETL service principal. Doing so requires the `Owner` role on your Azure account in the SBMTD subscription. (This has been added as a task on #297)

### CODE REVIEW

**Comment by angela-tran** [comment]
There's a little more code cleanup to do

File: terraform/variables.tf [hunk 1]
```diff
@@ -8,11 +8,6 @@ variable "AGENCY_CARD" {
   type        = string
 }
 
-variable "AGENCY_CARD_DATA_ETL_APP_OBJECT_ID" {
-  description = "Object ID from the registered application for the Agency Card server ETL uploading: https://cloudsight.zendesk.com/hc/en-us/articles/360016785598-Azure-finding-your-service-principal-object-ID"
-  type        = string
-}
-
 variable "AGENCY_CARD_DATA_ETL_FILE" {
```
Comment (thekaveman) [inline terraform/variables.tf:11]: I don't think we need this variable anymore either right?

**Review by thekaveman — CHANGES_REQUESTED** [review]

**Review by thekaveman — CHANGES_REQUESTED** [review]
And let's also delete the unused variables from the agency-specific `azure-vars.yml` files.

File: terraform/variables.tf (hunk 1 above)
Comment (angela-tran) [reply terraform/variables.tf]: Thanks for catching! I removed the variable and values in https://github.com/cal-itp/eligibility-server/pull/359/commits/83d5231d7f9510986f83ce3ed1de939c903c4683

**Comment by angela-tran** [comment]
Successfully ran the `terraform state rm...` command for MST prod workspace:
```
C:\dev\eligibility-server\terraform>terraform state rm azurerm_role_assignment.velocity_etl[0]
Removed azurerm_role_assignment.velocity_etl[0]
Successfully removed 1 resource instance(s).
```

**Review by thekaveman — APPROVED** [review]

**Pull request merged**